File: lib/faye/protocol/envelope.rb
module Faye
  class Envelope

    include Deferrable
    attr_reader :id, :message

    def initialize(message, timeout = nil)
      @id      = message['id']
      @message = message
      
      self.timeout(timeout, false) if timeout
    end

    def eql?(other)
      Envelope === other and @id == other.id
    end

    def hash
      @id.hash
    end

  end
end

File: javascript/protocol/envelope.js
Faye.Envelope = Faye.Class({
  initialize: function(message, timeout) {
    this.id      = message.id;
    this.message = message;

    if (timeout !== undefined) this.timeout(timeout / 1000, false);
  }
});

Faye.extend(Faye.Envelope.prototype, Faye.Deferrable);
